### PR TITLE
mono: add constants to transform and vector structs

### PR DESCRIPTION
**Pull request by KellyThomas**
**What this does:**

This PR brings the constants offered by `Transform`, `Transform2D`, `Vector2` and `Vector3` into line with GDScript changes in #14704

For the vectors I have opted for `Inf` over `Infinity` to match the `Mathf` property, I am happy to change this if the full name is preferred.

This swaps the values of `Vector2.Up`/`Vector2.Down`, so that they match the new GDScript implementation.

**What this doesn't do:**

I was unable add `X`, `Y`, or `Z` to `Plane` as it conflicts with existing instance properties.

I was unable to add `BlackCircle`, `WhiteCircle`, `BlackDiamond`, or `WhiteDiamond` to `StringExtensions` due to lack of support for static extension properties.

I haven't looked at the named colors.

### CODE REVIEW

**Review by aaronfranke — APPROVED** [review]
One nitpicky thing but otherwise it all looks good to me. :+1:

File: modules/mono/glue/cs_files/VERSION.txt [hunk 1]
```diff
@@ -1 +1 @@
-7
+9
```
[the remark this answers is not in the document]
Comment (KellyThomas) [reply modules/mono/glue/cs_files/VERSION.txt]: Yeah, that was probably a bit optimistic of me.

I've changed this back to 8 for now.

**Comment by akien-mga** [comment]
CC @neikeq

**Review by neikeq — CHANGES_REQUESTED** [review]

**Comment by KellyThomas** [comment]
Thanks, I've removed all alignment beyond the line indents.  It should be clean now.

**Review by neikeq — APPROVED** [review]

**Comment by akien-mga** [comment]
Now that Godot 3.1 has been released, we don't plan to cherry-pick new features and enhancements to the 3.0 branch, unless there is very strong support in favor of it. Removing cherrypick label for 3.0.

**Pull request merged**